sdk/core/azure-core/tests/test_messaging_cloud_event.py

```diff
@@ -469,4 +550,179 @@ def test_wrong_schema_raises_no_type():
+    cloud_storage_string = u'This is a random string which must fail'
+    obj = MockQueueMessage(content=cloud_storage_string)
+
+    with pytest.raises(ValueError, match="Failed to retrieve content from the object. Make sure the content follows the CloudEvent schema."):
+        _get_json_content(obj)
+
+def test_get_bytes_servicebus():
+    obj = MockServiceBusReceivedMessage(
+        body=MockBody(),
+        message_id='3f6c5441-5be5-4f33-80c3-3ffeb6a090ce',
+        content_type='application/cloudevents+json; charset=utf-8',
+        time_to_live=datetime.timedelta(days=14),
+        delivery_count=13,
+        enqueued_sequence_number=0,
+        enqueued_time_utc=datetime.datetime(2021, 7, 22, 22, 27, 41, 236000),
+        expires_at_utc=datetime.datetime(2021, 8, 5, 22, 27, 41, 236000),
+        sequence_number=11219,
+        lock_token='233146e3-d5a6-45eb-826f-691d82fb8b13'
+    )
+    dict = _get_json_content(obj)
+    assert dict.get('data') == "ServiceBus"
+    assert dict.get('specversion') == '1.0'
+
+def test_get_bytes_servicebus_wrong_content():
+    obj = MockServiceBusReceivedMessage(
+        body=MockBody(data="random string"),
+        message_id='3f6c5441-5be5-4f33-80c3-3ffeb6a090ce',
+        content_type='application/json; charset=utf-8',
+        time_to_live=datetime.timedelta(days=14),
+        delivery_count=13,
+        enqueued_sequence_number=0,
+        enqueued_time_utc=datetime.datetime(2021, 7, 22, 22, 27, 41, 236000),
+        expires_at_utc=datetime.datetime(2021, 8, 5, 22, 27, 41, 236000),
+        sequence_number=11219,
+        lock_token='233146e3-d5a6-45eb-826f-691d82fb8b13'
+    )
+
+    with pytest.raises(ValueError, match="Failed to retrieve body from the object. Make sure the body follows the CloudEvent schema."):
+        _get_json_content(obj)
+
+def test_get_bytes_eventhubs():
+    obj = MockEventhubData(
+        body=MockEhBody()
+    )
+    dict = _get_json_content(obj)
+    assert dict.get('data') == 'Eventhub'
+    assert dict.get('specversion') == '1.0'
+
+def test_get_bytes_eventhubs_wrong_content():
+    obj = MockEventhubData(
+        body=MockEhBody(data='random string')
+    )
+
+    with pytest.raises(ValueError, match="Failed to retrieve body from the object. Make sure the body follows the CloudEvent schema."):
+        dict = _get_json_content(obj)
+
+def test_get_bytes_random_obj():
+    json_str = '{"id": "de0fd76c-4ef4-4dfb-ab3a-8f24a307e033", "source": "https://egtest.dev/cloudcustomevent", "data": {"team": "event grid squad"}, "type": "Azure.Sdk.Sample", "time": "2020-08-07T02:06:08.11969Z", "specversion": "1.0"}'
+    random_obj =  {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e033",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2020-08-07T02:06:08.11969Z",
+        "specversion":"1.0"
+    }
+
+    assert _get_json_content(json_str) == random_obj
```

Comment:
Would be worth trying this with the full constructor call to ensure that "time" is deserialized into the timestamp.

Reply:
i've added these